src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2835,6 +2838,11 @@ void processContextElement
       responseP->contextElementResponseVector.push_back(cerP);
     }
   }
+  else if (apiVersion == "v2" && action == "APPEND" && !isUpdate)
```

Comment:
In a second thought, we probably need some other modifications to make it work. I would try to explain the rationale behind the needed modifications (but if my explanation is confusing, don't hesitate to ask for help in the Orion Dev skype chat :)

First, a modification is needed in the `POST /v2/entities` serviceRoutine to change APPEND to APPEND_STRICT in the mongoBackend invokation.

After that, we still having a problem, due to mongoUpdateContext() is not able to distinguish between the following situations:

* `POST /v2/entities`, which requires to check entity existence to solve #1158 
* `POST /v2/entities/<id>?op=append`, which doesn't required to check for entity existence as the operation my be used to add attributes to an existing entity.

Both operation are using APPEND_STRICT and the payload they use follow the same structure, so they cannot be distinguishes.

In order to solve this, we need to:

1. Add a new parameter to mongoUpdateContext() function signature, a parameter `bool checkEntityExistence` which default is `false`.
2. Set `checkEntityExistence` to `true` in `POST /v2/entities`
3. The #1158 error detection logic should be only done when APPEND_STRICT *and* checkEntityExistence are used

Anyway, first step is finishing the .test to cover the #1158 fix, as it is very valuable as guide to tune the implementation in the code.